test/cluster/security_test.go

```diff
@@ -0,0 +1,435 @@
+	require.NoError(t, err)
+	require.NotNil(t, config.LocalConfig)
+	require.Nil(t, config.SharedConfig)
+	require.NotNil(t, config.JoinBlock)
+
+	c.AddNewServerToCluster(server5)
+	require.EqualError(t, c.StartServer(server5), "failed to start the server: node-5")
+}
+
+// Scenario:
+// - start a 3 servers cluster with TLS enabled.
+// - change server-2 config file
+func TestTLSChangeNodeCA(t *testing.T) {
+	dir, err := ioutil.TempDir("", "int-test")
+	require.NoError(t, err)
+
+	nPort, pPort := getPorts(3)
+	setupConfig := &setup.Config{
+		NumberOfServers:     3,
+		TestDirAbsolutePath: dir,
+		BDBBinaryPath:       "../../bin/bdb",
+		CmdTimeout:          10 * time.Second,
+		BaseNodePort:        nPort,
+		BasePeerPort:        pPort,
+		ClusterTLSEnabled:   true,
+		CheckRedirectFunc: func(req *http.Request, via []*http.Request) error {
+			return errors.Errorf("Redirect blocked in test client: url: '%s', referrer: '%s', #via: %d", req.URL, req.Referer(), len(via))
+		},
+	}
+	c, err := setup.NewCluster(setupConfig)
+	require.NoError(t, err)
+	defer c.ShutdownAndCleanup()
+
+	//require.NoError(t, c.StartServer(c.Servers[0]))
+	//require.NoError(t, c.StartServer(c.Servers[1]))
+	require.NoError(t, c.Start())
+	s2 := c.Servers[2]
+	require.NoError(t, c.ShutdownServer(s2))
+
+	leaderIndex := -1
+	require.Eventually(t, func() bool {
+		leaderIndex = c.AgreedLeader(t, 0, 1)
+		return leaderIndex >= 0
+	}, 30*time.Second, 100*time.Millisecond)
+
+	certRootCA, rootCAPrivKey, err := testutils.GenerateRootCA("Orion RootCA", "127.0.0.1")
+	require.NoError(t, err)
+	require.NotNil(t, certRootCA)
+	require.NotNil(t, rootCAPrivKey)
+
+	keyPath, certPath, rootCACertPath, err := s2.CreateNewCryptoMaterials(certRootCA, rootCAPrivKey)
+	localConfig := &config.LocalConfiguration{
+		Replication: config.ReplicationConf{
+			TLS: config.TLSConf{
+				Enabled:               true,
+				ServerCertificatePath: certPath,
+				ServerKeyPath:         keyPath,
+				ClientCertificatePath: certPath,
+				ClientKeyPath:         keyPath,
+				CaConfig: config.CAConfiguration{
+					RootCACertsPath: []string{rootCACertPath},
+				},
+			},
+		},
+	}
+	require.NoError(t, s2.CreateConfigFile(localConfig))
+	require.NoError(t, c.StartServer(s2))
+
+	// submit tx
+	txID, rcpt, _, err := c.Servers[leaderIndex].WriteDataTx(t, worldstate.DefaultDBName, "key1", []byte("key1-data"))
+	require.NoError(t, err)
+	require.NotNil(t, rcpt)
+	require.True(t, txID != "")
+	require.True(t, len(rcpt.GetHeader().GetValidationInfo()) > 0)
+	require.Equal(t, types.Flag_VALID, rcpt.Header.ValidationInfo[rcpt.TxIndex].Flag)
+	t.Logf("tx submitted: %s, %+v", txID, rcpt)
+
+	// server2 is not sync with the last tx, because it is unable to connect to the cluster
+	dataEnv, err := c.Servers[2].QueryData(t, worldstate.DefaultDBName, "key1", "admin")
+	require.NoError(t, err)
+	require.NotNil(t, dataEnv)
+	require.Equal(t, []byte(nil), dataEnv.GetResponse().GetValue())
```

Comment:
we can also do an eventually that server 2 has no leader, like what you fixed on node removal, and that the leader has only 2 active